src/workerd/io/worker-entrypoint.c++

```diff
@@ -286,10 +290,22 @@ kj::Promise<void> WorkerEntrypoint::request(kj::HttpMethod method,
   TRACE_EVENT_BEGIN("workerd", "WorkerEntrypoint::request() waiting on context",
       PERFETTO_TRACK_FROM_POINTER(&context), PERFETTO_FLOW_FROM_POINTER(this));
 
+  // Using a DurationExceededLambda here instead of a DurationExceededLogger to avoid doing a string
+  // heap allocation in the happy path where the duration isn't exceeded.
```

Comment:
Rationale for introducing the DurationExceededLambda is here, but let me know if this rationale doesn't make sense, then I'll use the simpler DurationExceededLogger here. Just wanted to avoid heap allocations here since this is a hot path.